File: _cpp/answer/26.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size()==0)
            return 0;
        int k=0;

        for (int i = 1; i < nums.size(); ++i)
        {
            if (nums[i]!=nums[k]){
                nums[++k]=nums[i];
            }
        }

        return k+1;
    }
};

int main(int argc, char const *argv[])
{
    int arr[10]={1,2,2,2,3,3,3,3,3,4};
    vector<int> v;
    int res = Solution().removeDuplicates(v);
    cout<<"res:"<<res<<endl;

    for (int i = 0; i < v.size(); ++i)
    {
        cout<<v[i]<<" ";
    }
    return 0;
}

File: _python/26.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        k = 0
        for i in xrange(1, len(nums)):
            if nums[i] != nums[k]:
                k += 1
                nums[k] = nums[i]
        return k+1

    def s2(self,nums):
        k = 0
        for i in xrange(len(nums)):
            if k < 1 or nums[i] != nums[k - 1]:
                nums[k] = nums[i]
                k += 1
        return k

if __name__ == '__main__':
    nums = [1, 1, 1,2,2,2,2,2,3,4,5,5]
    print Solution().s2(nums)
    print nums
